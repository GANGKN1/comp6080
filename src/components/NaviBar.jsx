import React from 'react';
import { Link as RouterLink, NavLink, useHistory } from 'react-router-dom';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Button from '@mui/material/Button';
import { styled, useTheme } from '@mui/system';
import AllAPI from '../utils/AllAPI';
import { useSnackbar } from 'notistack';

const Great = styled(Typography)(({ theme }) => ({
  fontSize: '19pt',
  color: 'black',
  flexGrow: 1,
}));

const NavigationLink = styled(Button)(({ theme }) => ({
  color: 'black',
}));

const NaviBar = () => {
  const history = useHistory();
  const tempAPI = new AllAPI();
  const { enqueueSnackbar } = useSnackbar();
  const theme = useTheme();

  const marginBottomStyle = {
    marginBottom: theme.spacing(2),
  };

  // the func used for logging out
  const logOut = async () => {
    try {
      const token = localStorage.getItem('token');
      localStorage.removeItem('token');
      history.push('/');
      enqueueSnackbar('Logged out', { variant: 'success', autoHideDuration: 3000 });

      const response = await tempAPI.postAPIRequestToken('admin/auth/logout', token)
      const tempData = await response.json();
      if (!response.ok) {
        console.warn(tempData.message);
      }
    } catch (error) {
      console.warn(error);
    }
  }

  return (
    <AppBar position="static" color="default" sx={marginBottomStyle}>
      <Toolbar>
        <Great
          id="bigbrain-title"
          variant="h6"
          component={RouterLink}
          to="/dashboard"
          spacing={1}
        >
          BigBrain
        </Great>

        <NavigationLink
          id="bigbrain-dashboard"
          component={NavLink}
          to="/dashboard"
        >
          Dashboard
        </NavigationLink>

        <NavigationLink
          component={NavLink}
          id="bigbrain-logout"
          exact
          to="/"
          onClick={(event) => {
            event.preventDefault();
            logOut();
          }}
        >
          Log Out
        </NavigationLink>
      </Toolbar>
    </AppBar>
  );
};

export default NaviBar;
