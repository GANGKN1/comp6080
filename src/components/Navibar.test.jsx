import Enzyme, { shallow } from 'enzyme';
import React from 'react';
import NaviBar from './NaviBar';
import renderer from 'react-test-renderer';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { BrowserRouter as Router } from 'react-router-dom';
Enzyme.configure({ adapter: new Adapter() });

describe('NaviBar', () => {
  const createNaviBar = () => renderer.create(<Router><NaviBar /></Router>).toJSON();
  const shallowNaviBar = () => shallow(<NaviBar />);

  it('There is a BigBrain Title', () => {
    const step1 = createNaviBar();
    const step2 = shallowNaviBar();
    const text = step2.find('#bigbrain-title');

    expect(step2).toHaveLength(1);
    expect(text).toHaveLength(1);
    expect(text.text()).toBe('BigBrain');
    expect(step1).toMatchSnapshot();
  });

  it('There is a dashboard button for clicking', () => {
    const step1 = createNaviBar();
    const step2 = shallowNaviBar();
    const text = step2.find('#bigbrain-dashboard');

    expect(text.text()).toBe('Dashboard');
    expect(step1).toMatchSnapshot();
  });

  it('There is a logout button for clicking', () => {
    const step1 = createNaviBar();
    const step2 = shallowNaviBar();
    const text = step2.find('#bigbrain-logout');

    expect(text.text()).toBe('Log Out');
    expect(step1).toMatchSnapshot();
  });
});
