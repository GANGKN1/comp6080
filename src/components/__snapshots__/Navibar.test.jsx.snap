// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NaviBar There is a BigBrain Title 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionStatic css-vb9rq3-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <a
      className="MuiTypography-root MuiTypography-h6 css-p80pko-MuiTypography-root"
      href="/dashboard"
      id="bigbrain-title"
      onClick={[Function]}
      spacing={1}
    >
      BigBrain
    </a>
    <a
      aria-current={null}
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-148uago-MuiButtonBase-root-MuiButton-root"
      href="/dashboard"
      id="bigbrain-dashboard"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Dashboard
    </a>
    <a
      aria-current="page"
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-148uago-MuiButtonBase-root-MuiButton-root active"
      href="/"
      id="bigbrain-logout"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      tabIndex={0}
    >
      Log Out
    </a>
  </div>
</header>
`;

exports[`NaviBar There is a dashboard button for clicking 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionStatic css-vb9rq3-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <a
      className="MuiTypography-root MuiTypography-h6 css-p80pko-MuiTypography-root"
      href="/dashboard"
      id="bigbrain-title"
      onClick={[Function]}
      spacing={1}
    >
      BigBrain
    </a>
    <a
      aria-current={null}
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-148uago-MuiButtonBase-root-MuiButton-root"
      href="/dashboard"
      id="bigbrain-dashboard"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Dashboard
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      aria-current="page"
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-148uago-MuiButtonBase-root-MuiButton-root active"
      href="/"
      id="bigbrain-logout"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      tabIndex={0}
    >
      Log Out
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</header>
`;

exports[`NaviBar There is a logout button for clicking 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionStatic css-vb9rq3-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <a
      className="MuiTypography-root MuiTypography-h6 css-p80pko-MuiTypography-root"
      href="/dashboard"
      id="bigbrain-title"
      onClick={[Function]}
      spacing={1}
    >
      BigBrain
    </a>
    <a
      aria-current={null}
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-148uago-MuiButtonBase-root-MuiButton-root"
      href="/dashboard"
      id="bigbrain-dashboard"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Dashboard
    </a>
    <a
      aria-current="page"
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-148uago-MuiButtonBase-root-MuiButton-root active"
      href="/"
      id="bigbrain-logout"
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      tabIndex={0}
    >
      Log Out
    </a>
  </div>
</header>
`;
